contribute/design/1.x/cstor-operator/doc.md

```diff
@@ -46,107 +46,154 @@ Following is the proposed CSPC schema in go struct:
+	// If tolerations at PoolConfig is empty, this is written to
+	// CSPI PoolConfig.
+	Tolerations []corev1.Toleration `json:"tolerations"`
+
+	// DefaultPriorityClassName if specified applies to all the pool pods
+	// in the pool spec if the priorityClass at the pool level is
+	// not specified.
+	DefaultPriorityClassName string `json:"priorityClassName"`
 }
 
-//PoolSpec is the spec for a pool on a node where it should be created.
+//PoolSpec is the spec for pool on node where it should be created.
 type PoolSpec struct {
-  // NodeSelector is the labels that will be used to select
-  // a node for pool provisioning.
-  // Required field
-  NodeSelector map[string]string `json:"nodeSelector"`
-  // RaidConfig is the raid group configuration for the given pool.
-  RaidGroups []RaidGroup `json:"raidGroups"`
-  // PoolConfig is the default pool config that applies to the
-  // pool on node.
-  PoolConfig PoolConfig `json:"poolConfig"`
+	// NodeSelector is the labels that will be used to select
+	// a node for pool provisioning.
+	// Required field
+	NodeSelector map[string]string `json:"nodeSelector"`
+
+	// DataRaidConfig is the raid group configuration for the given pool.
+	DataRaidGroups []RaidGroup `json:"dataRaidGroups"`
+
+	// WriteCacheGroups is the write cache given pool.
+	WriteCacheGroups []RaidGroup `json:"writeCacheGroups"`
+
+	// PoolConfig is the default pool config that applies to the
+	// pool on node.
+	PoolConfig PoolConfig `json:"poolConfig"`
 }
 
 // PoolConfig is the default pool config that applies to the
 // pool on node.
 type PoolConfig struct {
-  // Cachefile is used for faster pool imports
-  // optional -- if not specified or left empty cache file is not
-  // used.
-  CacheFile string `json:"cacheFile"`
-  // DefaultRaidGroupType is the default raid type which applies
-  // to all the pools if raid type is not specified there
-  // Compulsory field if any raidGroup is not given Type
-  DefaultRaidGroupType string `json:"defaultRaidGroupType"`
-
-  // OverProvisioning to enable over provisioning
-  // Optional -- defaults to false
-  OverProvisioning bool `json:"overProvisioning"`
-  // Compression to enable compression
-  // Optional -- defaults to off
-  // Possible values : lz, off
-  Compression string `json:"compression"`
+	// DataRaidGroupType is the data raid group raid type 
+	// Supported values are : stripe, mirror, raidz and raidz2
+	DataRaidGroupType string `json:"dataRaidGroupType"`
+
+	// WriteCacheRaidGroupType is the write cachce raid type 
+	// Supported values are : stripe, mirror, raidz and raidz2
+	WriteCacheRaidGroupType string `json:"writeCacheRaidGroupType"`
+
+	// ThickProvisioning to enable over thick provisioning
+	// Optional -- defaults to false
+	ThickProvisioning bool `json:"thickProvisioning"`
+
+	// Compression to enable compression
+	// Optional -- defaults to off
+	// Possible values : lz, off
+	Compression string `json:"compression"`
```

Comment:
What does Compression do at a pool level? Its set at a volume level usually.

Reply:
I think it might be added for reason that volumes on that pool will **inherit this property from pool** in case if volumes properties don't say any info about compression. cc: @vishnuitta correct me If I am wrong on it.

Reply:
If compression is set at the pool level, and it is not given while provisioning the volume, then it will be on as it will inherit that from the pool. However, if you are specifically mentioning the compression at the volume level then that will be considered.